runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -110,14 +110,25 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
   return vim.tbl_map(function(diagnostic)
     local start = diagnostic.range.start
     local _end = diagnostic.range['end']
+    local message ---@type string
+    if type(diagnostic.message) == 'string' then
+      ---@cast diagnostic { message: string }
+      message = diagnostic.message
+    else
+      vim.notify_once(
+        string.format('Unsupported Markup message from LSP client %d', client_id),
+        vim.lsp.log_levels.ERROR
+      )
+      message = diagnostic.message.value
+    end
```

Comment:
```suggestion
    local message = diagnostic.message
    if type(message) ~= 'string' then
      vim.notify_once(
        string.format('Unsupported Markup message from LSP client %d', client_id),
        vim.lsp.log_levels.ERROR
      )
      message = message.value
    end
```

To reflect the remark from @lewis6991

Reply:
@mfussenegger done :)